src/core/createConfig.js

```diff
@@ -118,3 +129,11 @@ export const required = (config, key, currentValue) => {
   }
   return err;
 };
+export const matchesRegEx = ex => (config, key, currentValue) => {
```

Comment:
Couple of things:

1. This validator is too low level; no real point of a helper if we'll be asking the components' devs to bring in their own regex. Let's make this a private function and use it to expose a `matchUrl` validator.

`const matchUrl = createRegexValidator(regex);`

2. Since we have more than 1 validator now, let's move those bad boys out of here into their own cozy `validators` folder.

Reply:
I removed the regex validator (for now), and added a specific one for validating the secure URL. I also created a .. cozy validation folder that can host the validation we need going forward.